src/services/publisher_service/mod.rs

```diff
@@ -177,6 +241,33 @@ async fn process_queued_messages(
     Ok(())
 }
 
+/// Process publishing with the threshold timeout
+#[instrument(skip(relay_http_client, metrics, analytics, notification))]
+async fn process_with_timeout(
+    execution_threshold: Duration,
+    notification: NotificationToProcess,
+    relay_http_client: Arc<Client>,
+    metrics: Option<&Metrics>,
+    analytics: &NotifyAnalytics,
+) -> crate::error::Result<()> {
+    match timeout(
```

Comment:
Interesting you added a timeout here. I wouldn't expect this would be required because we only retry up to 10 times and the retries themselves shouldn't take more than HTTP timeout so that would be what, max 20 minutes? Doesn't hurt to add I guess.

Reply:
As this timeout (when the message is considered as a dead letter) can be changed we should change the HTTP timeouts as well to eliminate cases where the HTTP requests are too long and the message will be considered as a dead letter when the worker is actually processing it. So timeout here should solve this possible case.